### PR TITLE
Sam/pgroonga deps

### PR DESCRIPTION
**Pull request by samrose**
## What kind of change does this PR introduce?

a comment on a PR https://github.com/supabase/postgres/pull/462#issuecomment-2270662752 reported that in our 15.6 release, `TokenMecab` was no longer working on the `pgroonga` extension.

This PR restores that Tokenizer capability to the proonga extension, and adds the following test below, which will run on every PR going forward. 

```sql
-- File: 0005-test_pgroonga_revised.sql

begin;
    -- Plan for 3 tests: extension, table, and index
    select plan(3);
    
    -- Create the PGroonga extension
    create extension if not exists pgroonga;
    
    -- -- Test 1: Check if PGroonga extension exists
    select has_extension('pgroonga', 'The pgroonga extension should exist.');
    
    -- Create the table
    create table notes(
        id integer primary key,
        content text
    );
    
    -- Test 2: Check if the table was created
    SELECT has_table('public', 'notes', 'The notes table should exist.');    
    -- Create the PGroonga index
    CREATE INDEX pgroonga_content_index
            ON notes
         USING pgroonga (content)
          WITH (tokenizer='TokenMecab');
    
    -- -- Test 3: Check if the index was created
    SELECT has_index('public', 'notes', 'pgroonga_content_index', 'The pgroonga_content_index should exist.');
    
    -- -- Cleanup (this won't affect the test results as they've already been checked)
    DROP INDEX IF EXISTS pgroonga_content_index;
    DROP TABLE IF EXISTS notes;
    
    -- Finish the test plan
    select * from finish();
rollback;
```

### CODE REVIEW

File: nix/ext/pgroonga.nix
Comment (darora) [inline nix/ext/pgroonga.nix]: Is there an existing log file? Is something set up to handle rotation for it?

**Review by darora — APPROVED** [review]

File: nix/ext/pgroonga.nix
Comment (samrose) [reply nix/ext/pgroonga.nix]: @darora good point, I don't think there is an existing logfile in production.

I actually turned this on for debugging purposes, and will definitely remove it before merge

Comment (samrose) [reply nix/ext/pgroonga.nix]: If we decided we wanted that on, there may be a better way to do it post-build, but likely we don't need that logging is my guess.

Comment (samrose) [reply nix/ext/pgroonga.nix]: @div I remove this and the other sql patching

**Review by olirice — APPROVED** [review]

**Comment by olirice** [comment]
@samrose lgtm, feel free to merge once the conflicts are resolved

**Pull request merged**
